### PR TITLE
[docs] update k8s helm docs

### PR DESCRIPTION
**Pull request by kschoche**
### Description
Describe why you're making this change, in plain English.

Autogenerate docs based on consul-k8s commit `d3a702df2180cce97a7c34a68470639b70493bba` which corresponds to our 0.44.0 release.

### Testing & Reproduction steps
* In the case of bugs, describe how to replicate
* If any manual tests were done, document the steps and the conditions to replicate
* Call out any important/ relevant unit tests, e2e tests or integration tests you have added or are adding

### Links
Include any links here that might be helpful for people reviewing your PR (Tickets, GH issues, API docs, external benchmarks, tools docs, etc). If there are none, feel free to delete this section.

Please be mindful not to leak any customer or confidential information. HashiCorp employees may want to use our internal URL shortener to obfuscate links.

### PR Checklist

* [ ] updated test coverage
* [ ] external facing docs updated
* [ ] not a security concern

### CODE REVIEW

**Review by lkysow — APPROVED** [review]

**Review by trujillo-adam — CHANGES_REQUESTED** [review]
All of the the links to agent configuration options are now incorrect. For example, https://consul.io/docs/agent/config/cli-flags#_domain was changed to https://consul.io/docs/agent/options#_domain, but these were all updated per #12005

**Comment by kschoche** [comment]
> All of the the links to agent configuration options are now incorrect. For example, https://consul.io/docs/agent/config/cli-flags#_domain was changed to https://consul.io/docs/agent/options#_domain, but these were all updated per #12005

Ah, I thought it was the other way around.
This doc is auto-generated, so that means `hashicorp/consul-k8s-control-plane` is actually referencing incorrect links. I can submit a PR there to fix those links, and then come back to this one.
